### ISSUE TITLE
Create endpoints for linking and unlinking Data Requests With Locations

### ISSUE THREAD

**Issue by maxachis** [Police-Data-Accessibility-Project/data-sources-app#505]
### Context

Currently, in the `/data-requests` `POST` endpoint, we include a parameter 

### Requirements

1. Add `/data-requests/{resource_id}/related-locations` `GET`, `POST`, and `DELETE` endpoints, for obtaining, adding, and removing locations, respectively, similar in behavior to our existing logic for `/data-sources/{resource_id}/related-agencies`
2. Additionally, update `/typeahead/locations` to include the location id of a given location, for easy connection of a location id

### Tests

* 1 will require entirely new integration tests to guage the logic
* 2 will require an update to the existing `/typeahead/locations` tests to check for the presence of the correct location ID

### Docs

* API documentation will need updated for 1 and 2

### Open questions

* None at this time.

**Comment by maxachis** [comment]
@josh-chamberlain Do we want to be able to allow users to update their locations after creating their data requests?

**Comment by josh-chamberlain** [comment]
@maxachis yes, I think that's a good idea. That said, we don't have a UI for letting users edit their requests—so for now, the only people doing that would be those editing the requests via retool.

**Comment by maxachis** [comment]
> @maxachis yes, I think that's a good idea. That said, we don't have a UI for letting users edit their requests—so for now, the only people doing that would be those editing the requests via retool.

In that case, (as you've already seen in Retool) we've resolved everything there is to resolve in this issue! 🎉
